clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3177,25 +3177,32 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 
   ASTContext &Ctx = S.getASTContext();
 
-  if (!XDimExpr->isValueDependent() && !YDimExpr->isValueDependent() &&
-      !ZDimExpr->isValueDependent()) {
-    llvm::APSInt XDimVal, YDimVal, ZDimVal;
-    ExprResult XDim = S.VerifyIntegerConstantExpression(XDimExpr, &XDimVal);
-    ExprResult YDim = S.VerifyIntegerConstantExpression(YDimExpr, &YDimVal);
-    ExprResult ZDim = S.VerifyIntegerConstantExpression(ZDimExpr, &ZDimVal);
+  bool YDimExprIsDereferencable = YDimExpr && (!YDimExpr->isValueDependent());
+  bool ZDimExprIsDereferencable = ZDimExpr && (!ZDimExpr->isValueDependent());
 
+  llvm::APSInt XDimVal, YDimVal, ZDimVal;
+  if (!XDimExpr->isValueDependent()) {
+    ExprResult XDim = S.VerifyIntegerConstantExpression(XDimExpr, &XDimVal);
     if (XDim.isInvalid())
       return;
     XDimExpr = XDim.get();
+  }
 
+  if (YDimExprIsDereferencable) {
+    ExprResult YDim = S.VerifyIntegerConstantExpression(YDimExpr, &YDimVal);
     if (YDim.isInvalid())
       return;
     YDimExpr = YDim.get();
+  }
 
+  if (ZDimExprIsDereferencable) {
+    ExprResult ZDim = S.VerifyIntegerConstantExpression(ZDimExpr, &ZDimVal);
     if (ZDim.isInvalid())
       return;
     ZDimExpr = ZDim.get();
+  }
 
+  if ((XDimVal >= 1) && (YDimVal >= 1) && (ZDimVal >= 1)) {
```

Comment:
Why are you checking for `>= 1` now?

Reply:
I did that instead of checking that `(XDimExprIsDereferencable && YDimExprIsDereferencable && ZDimExprIsDereferencable) == 1` Because `XDimVal`, `YDimVal` and `ZDimVal` are definitely >= 1 in cases where `YDimExpr` and `ZDimExpr` are not `nullptr`

Reply:
Okay, I think I see what you're doing now, thanks! The stuff you are guarding can only be called if we have the x, y, and z dims available (they're not value dependent), so you're sort of trying to check the same thing, but in a different way than before. I think the old approach was a bit more clear -- it's not that we care about the value itself, it's that we care there *is* a value.

Reply:
I don't think you can count on the value here, the user could insert a '1' here as well.

Reply:
Ok. Will change it back to the old approach. Thanks!